src/test/java/uk/gov/justice/digital/delius/controller/secure/CustodyKeyDatesControllerTest.java

```diff
@@ -0,0 +1,136 @@
+package uk.gov.justice.digital.delius.controller.secure;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.fasterxml.jackson.datatype.jdk8.Jdk8Module;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import io.restassured.module.mockmvc.RestAssuredMockMvc;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import uk.gov.justice.digital.delius.controller.advice.SecureControllerAdvice;
+import uk.gov.justice.digital.delius.data.api.Custody;
+import uk.gov.justice.digital.delius.data.api.CustodyRelatedKeyDates;
+import uk.gov.justice.digital.delius.data.api.ReplaceCustodyKeyDates;
+import uk.gov.justice.digital.delius.service.ConvictionService;
+import uk.gov.justice.digital.delius.service.OffenderService;
+
+import java.util.Optional;
+
+import static io.restassured.module.mockmvc.RestAssuredMockMvc.given;
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
+
+class CustodyKeyDatesControllerTest {
+
+    private OffenderService offenderService = mock(OffenderService.class);
+    private ConvictionService convictionService = mock(ConvictionService.class);
+    private ObjectMapper objectMapper = new ObjectMapper()
+            .configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
+            .configure(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS, false)
+            .setSerializationInclusion(JsonInclude.Include.NON_NULL)
+            .setSerializationInclusion(JsonInclude.Include.NON_ABSENT)
+            .setSerializationInclusion(JsonInclude.Include.NON_EMPTY)
+            .registerModules(new Jdk8Module(), new JavaTimeModule());
```

Comment:
Is this the right place to configure the object mapper?  It might get out of sync with the real object mapper.  (In the past I have configured object mappers in a static method just so I can reuse in unit tests)

Reply:
Totally agree, I'll do something next time around. The problem being is we don't use Spring to inject this so we need one that matches what we do in Spring